week-3/Homework/G-extra/1-map.js

```diff
@@ -10,10 +10,24 @@ var stationTransportOptionsPairs = [
   ["Angel", ["tube", "bus"]],
   ["London Bridge", ["tube", "bus", "river boat"]],
   ["Tower bridge", ["tube", "bus"]],
-  ["Greenwich", ["tube", "bus", "river boat"]]
+  ["Greenwich", ["tube", "bus", "river boat"]],
 ];
+function riverBoatFiltered(pair) {
+  let station = pair[0];
+  let transport = pair[1];
+  // console.log(station);
+  // console.log(transport);
+  if (transport.includes("river boat")) {
+    return station;
+  }
+  return;
+}
 
-var stationsWithRiverBoat; // <-- Complete this statement
+var stationsWithRiverBoat = stationTransportOptionsPairs
+  .map(riverBoatFiltered)
+  .filter((station) => {
+    return station !== undefined;
+  }); // <-- Complete this statement
 
```

Comment:
This works but it's not really what's being looked for. The `.filter()` function is intended for applying a filter, whilst `.map()` is in tended for doing something based on each value of the array, such as only returning part of it. It's also expected that the function with which you are filtering return explicit `true` or `false` values. So, for example, whether it's `true` or `false` that transport includes river boat. Then you can map the filtered values to the names of the stations.

Reply:
Thanks for your feedback. Yes, I have changed the code and committed it. It became shorter and more clear. Thanks again.